velox/functions/prestosql/tests/ZipTest.cpp

```diff
@@ -0,0 +1,222 @@
+      {{1, 2, 3, 4}, {3, 4, 5}, {std::nullopt}});
+  auto secondVector =
+      makeArrayVector<int64_t>({{2, 2, 2}, {1, 1, 1}, {3, 3, 3}});
+
+  auto result = evaluate<ArrayVector>(
+      "zip(c0, c1)",
+      makeRowVector({
+          firstVector,
+          secondVector,
+      }));
+
+  auto firstResult = makeNullableFlatVector<int64_t>(
+      {1, 2, 3, 4, 3, 4, 5, std::nullopt, std::nullopt, std::nullopt});
+  auto secondResult = makeNullableFlatVector<int64_t>(
+      {2, 2, 2, std::nullopt, 1, 1, 1, 3, 3, 3});
+
+  auto rowVector = makeRowVector({firstResult, secondResult});
+
+  // create the expected ArrayVector
+  auto expected = createArrayVector({0, 4, 7}, {4, 3, 3}, rowVector);
+
+  assertEqualVectors(expected, result);
+}
+
+/// Test if we can zip across different data types.
+TEST_F(ZipTest, combineInt) {
+  using S = StringView;
+  auto firstVector = makeArrayVector<int64_t>({{1, 1, 1, 1}, {2, 2, 2}, {}});
+  auto secondVector = makeArrayVector<StringView>(
+      {{S("a"), S("a"), S("a")}, {S("b"), S("b"), S("b")}, {S("c"), S("c")}});
+
+  auto result = evaluate<ArrayVector>(
+      "zip(c0, c1)",
+      makeRowVector({
+          firstVector,
+          secondVector,
+      }));
+
+  auto firstResult = makeNullableFlatVector<int64_t>(
+      {1, 1, 1, 1, 2, 2, 2, std::nullopt, std::nullopt, std::nullopt});
+  auto secondResult = makeNullableFlatVector<std::string>(
+      {S("a"),
+       S("a"),
+       S("a"),
+       std::nullopt,
+       S("b"),
+       S("b"),
+       S("b"),
+       S("c"),
+       S("c")});
+
+  auto rowVector = makeRowVector({firstResult, secondResult});
+
+  // create the expected ArrayVector
+  auto expected = createArrayVector({0, 4, 7}, {4, 3, 2}, rowVector);
+
+  assertEqualVectors(expected, result);
+}
+
+/// Test if we can zip with vectors containing null and empty Arrays
+TEST_F(ZipTest, nullEmptyArray) {
+  auto O = [](std::vector<std::optional<int32_t>> data) {
+    return std::make_optional(data);
+  };
+
+  auto firstVector =
+      makeVectorWithNullArrays<int32_t>({O({1, 1, 1, 1}), O({}), std::nullopt});
+
+  auto secondVector = makeArrayVector<int32_t>({{0, 0, 0}, {4, 4}, {5, 5}});
+
+  auto result = evaluate<ArrayVector>(
+      "zip(c0, c1)",
+      makeRowVector({
+          firstVector,
+          secondVector,
+      }));
+
+  auto firstResult =
+      makeNullableFlatVector<int32_t>({1, 1, 1, 1, std::nullopt, std::nullopt});
+  auto secondResult =
+      makeNullableFlatVector<int32_t>({0, 0, 0, std::nullopt, 4, 4});
+
+  auto rowVector = makeRowVector({firstResult, secondResult});
+
+  // create the expected ArrayVector
+  auto expected = createArrayVector({0, 4, 6}, {4, 2, 0}, rowVector);
+
+  assertEqualVectors(expected, result);
+}
+
+/// Test if we can pass multiple different array types in one function.
+TEST_F(ZipTest, arity) {
+  using S = StringView;
+  auto firstVector = makeArrayVector<int64_t>({{1, 1, 1, 1}, {2, 2, 2}, {}});
```

Comment:
Assuming we want to test 4 different types, shouldn't this be int16_t?

Reply:
This test was to make sure that i can pass more than 2 arrays and not get a resolver error. But i will make them different types.